modules/apps/headless/headless-builder/headless-builder-test/src/testFunctional/macros/ApplicationAPI.macro

```diff
@@ -0,0 +1,108 @@
+definition {
+
+	macro _curlAPIApplication {
+		if (isSet(virtualHost)) {
+			if (!(isSet(port))) {
+				var port = 8080;
+			}
+
+			var portalURL = "http://${virtualHost}:${port}";
+		}
+		else {
+			var portalURL = JSONCompany.getPortalURL();
+		}
+
+		if (isSet(aPIApplicationId)) {
+			var api = "headless-builder/applications/${aPIApplicationId}";
+		}
+		else {
+			var api = "headless-builder/applications";
+		}
+
+		var curl = '''
+			${portalURL}/o/${api} \
+				-u test@liferay.com:test \
+				-H Content-Type: application/json
+		''';
+
+		return ${curl};
+	}
+
+	macro createAPIApplication {
+		Variables.assertDefined(parameterList = "${baseURL},${statusKey},${statusName},${title}");
+
+		var curl = ApplicationAPI._curlAPIApplication();
+		var body = '''
+			-d {
+				"applicationStatus": {"key": "${statusKey}","name": "${statusName}"},
+				"baseURL": "${baseURL}",
+				"title": "${title}"
+			}
+		''';
+
+		var curl = StringUtil.add(${curl}, " \ ${body}", "");
+
+		var response = JSONCurlUtil.post(${curl});
+
+		return ${response};
+	}
+
+	macro deleteAllAPIApplication {
+		var getAPIApplicationList = ApplicationAPI.getAPIApplications();
+
+		var aPIApplicationIdList = JSONPathUtil.getProperty(
+			property = "$.items[*].id",
+			response = ${getAPIApplicationList});
+
+		for (var aPIApplicationIdList : list ${aPIApplicationIdList}) {
+			ApplicationAPI.deleteAPIApplicationById(aPIApplicationId = ${aPIApplicationIdList});
+		}
+
+		return ${response};
+	}
+
+	macro deleteAPIApplicationById {
+		Variables.assertDefined(parameterList = ${aPIApplicationId});
+
+		var curl = ApplicationAPI._curlAPIApplication(aPIApplicationId = ${aPIApplicationId});
+
+		var response = JSONCurlUtil.delete(${curl});
+
+		return ${response};
+	}
+
+	macro getAPIApplications {
+		var curl = ApplicationAPI._curlAPIApplication();
+
+		var response = JSONCurlUtil.get(${curl});
+
+		return ${response};
+	}
+
+	macro updateAPIApplication {
+		Variables.assertDefined(parameterList = "${aPIApplicationId},${parameter},${parameterValue}");
+
+		var curl = ApplicationAPI._curlAPIApplication(
+			aPIApplicationId = ${aPIApplicationId},
+			parameter = ${parameter},
+			parameterValue = ${parameterValue});
+
+		if (${parameter} == "applicationStatus") {
```

Comment:
Just wondering is there a way we can remove this and set the "" for parameterValue if needs in the method where we call it in .testcase file?

Reply:
I mean we already have the following in your testcase file, maybe we can remove the if/else and include one more "" in the following code?
```parameterValue = "{\"key\": \"unpublished\",\"name\": \"unpublished\"}");

Reply:
Thanks, with the [reply in the Slack](https://liferay.slack.com/archives/C03U66A6NNN/p1690519651045849) channel, we can modify it.